llvm/lib/CodeGen/ExpandLargeDivRem.cpp

```diff
@@ -54,8 +54,34 @@ static bool isSigned(unsigned int Opcode) {
   return Opcode == Instruction::SDiv || Opcode == Instruction::SRem;
 }
 
+static void scalarize(BinaryOperator *BO,
+                      SmallVectorImpl<BinaryOperator *> &Replace) {
+  VectorType *VTy = cast<VectorType>(BO->getType());
+  assert(!VTy->isScalableTy() && "Tried to scalarize scalable vector!");
```

Comment:
What's the difficulty with just directly handling vectors?

Reply:
The actual expansion routine in IntegerDivision.cpp can't handle them.

```
assert(!Div->getType()->isVectorTy() && "Div over vectors not supported");
```

The best solution would be to teach it how to do it, but I think it gets a bit messy with vectors since there is control flow involved in all of these expansions.